applets/ctap/ctap.c

```diff
@@ -497,11 +525,16 @@ static uint8_t ctap_get_assertion(CborEncoder *encoder, uint8_t *params, size_t
   // signature
   ret = cbor_encode_int(&map, RESP_signature);
   CHECK_CBOR_RET(ret);
-  sha256_init();
-  sha256_update(data_buf, len);
-  sha256_update(ga.clientDataHash, sizeof(ga.clientDataHash));
-  sha256_final(data_buf);
-  len = sign_with_private_key(pri_key, data_buf, data_buf);
+  if (rk.credential_id.alg_type == COSE_ALG_ES256) {
+    sha256_init();
+    sha256_update(data_buf, len);
+    sha256_update(ga.clientDataHash, sizeof(ga.clientDataHash));
+    sha256_final(data_buf);
+    len = sign_with_ecdsa_private_key(pri_key, data_buf, data_buf);
+  } else if (rk.credential_id.alg_type == COSE_ALG_EDDSA) {
+    memcpy(data_buf + len, ga.clientDataHash, CLIENT_DATA_HASH_SIZE);
```

Comment:
Potentially avoid this copy?

Reply:
Also note buffer size above if copy is avoided

Reply:
Nop, `ed25519_sign` can only sign one continuous buffer, to concat, we must use `memcpy`. Unless we write another function with this special capability (rely on `sha512_update`)